app/services/campaign_updater.rb

```diff
@@ -7,17 +7,38 @@ def self.run(params)
   end
 
   def initialize(params)
-    @params = params.slice(:name)
+    @params = params
     @campaign_id = params[:campaign_id] || raise(ArgumentError)
+    @retry_count = 0
   end
 
   def run
     ak_campaign_uri = CampaignRepository.get!(@campaign_id)
     ak_campaign_id = ActionKitConnector::Util.extract_id_from_resource_uri(ak_campaign_uri)
 
-    client.update_multilingual_campaign(ak_campaign_id, @params).tap do |response|
-      if !response.success?
-        raise Error.new("HTTP Response code: #{response.code}, body: #{response.body}")
+    response = client.update_multilingual_campaign(ak_campaign_id, sanitized_params)
+    if response.success?
+      response
+    elsif name_uniqueness_error?(response)
+      @retry_count += 1
+      run
+    else
+      raise Error.new("HTTP Response code: #{response.code}, body: #{response.body}")
+    end
+  end
+
+  private
+
+  #DUP name-uniqueness
```

Comment:
I can extract the `name_uniqueness_error?` into its own module. But I think doing the same for the `sanitize_params` method would not be a good solution. The new module code and the service class code would be very coupled and would make the logic a lot harder to understand. Specially around how the `@retry_count` is handled. So, IMO, the code duplication in that case is less bad than extracting that into a module. If you have another idea on how I can extract that I'm open to suggestions.